extensions/skins/conversation_v1/Conversation.js

```diff
@@ -646,3 +646,41 @@ oppia.directive('progressDots', [function() {
     }]
   };
 }]);
+oppia.filter('sliceDots', function() {
```

Comment:
I don't think this should be a separate filter. Can the behaviour be folded into the progressDots directive?

The code in the filter also looks fairly complex. It might be cleaner to keep track of just leftmostVisibleDotIndex and rightmostVisibleDotIndex in the JS. Then, in the HTML, show a dot only if it's within this range, and similarly use these variables as the basis for the fadeout treatment, arrow disabling, etc.